### PR TITLE
[SDK] Perf improvements with Promise.all

### PR DESCRIPTION
**Pull request by kien-ngo**
## Don't merge this
I'm aware this is a lot of changes. I will probably split it up into smaller PRs so that we have more control

## Problem solved

We have a lot of Promise.all that are not calling the inner requests concurrently, rather sequencially.

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [x] Internal API changes: explain the internal logic changes

## How to test

- [x] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

Self note: The following Pormise.all in this file can be improved however it seems too complex dont wanna touch that rn
- [ ] packages\sdk\src\evm\core\classes\drop-erc1155-claim-conditions.ts
- [ ] packages\sdk\src\evm\common\claim-conditions\processClaimConditionInputs.ts
- [ ] packages\sdk\src\evm\common\any-evm-utils\computeDeploymentInfo.ts

### CODE REVIEW

File: packages/sdk/src/evm/common/legacy.ts [hunk 1]
```diff
@@ -7,12 +7,10 @@ export async function getPrebuiltInfo(
...
-    const [type, version] = await Promise.all([
-      utils
-        .toUtf8String(await contract.contractType()) // eslint-disable-next-line no-control-regex
-        .replace(/\x00/g, ""),
-      await contract.contractVersion(),
-    ]);
+    const version = await contract.contractType();
+    const type = utils
+      .toUtf8String(version) // eslint-disable-next-line no-control-regex
+      .replace(/\x00/g, "");
```
Comment (jnsdls) [inline packages/sdk/src/evm/common/legacy.ts:13]: this looks potentially wrong - version used to be `contractVersion()` and now became `contractType()`?

Comment (kien-ngo) [reply packages/sdk/src/evm/common/legacy.ts:13]: oops I'll fix it

Comment (kien-ngo) [reply packages/sdk/src/evm/common/legacy.ts]: https://github.com/thirdweb-dev/js/pull/1648/commits/6d0588efeee56f83b1845b157e712f2a4baeb2ab

**Comment by kien-ngo** [comment]
Self note: The following files are to be cleaned up/improved
image

**Comment by kien-ngo** [comment]
Most changes are published. Happy to close this

**Pull request closed without merging**